### ISSUE TITLE
Unable to make a function conform to an overloaded callable protocol

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#1276]
## Environment data

-   Language Server version: 2021.5.1
-   OS and version: Windows 7
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.8 Anaconda

## Sample

```python
from __future__ import annotations

from typing import Any, Protocol, overload

class P(Protocol):
    @overload
    def __call__(self, x: int, /) -> int:
        ...

    @overload
    def __call__(self, x: str, /) -> str:
        ...

    def __call__(self, x: Any, /) -> Any:
        ...

def f0(x: Any, /) -> Any:
    return x

a: P = f0

@overload
def f1(x: int, /) -> int:
    ...

@overload
def f1(x: str, /) -> str:
    ...

def f1(x: Any, /) -> Any:
    return x

b: P = f1
```

## Expected behavior

No errors/warnings.

## Actual behaviour

Both assignments got warnings.

![image](https://user-images.githubusercontent.com/15389744/118097010-aff6fb80-b404-11eb-8714-aba1308fbe14.png)

![image](https://user-images.githubusercontent.com/15389744/118097036-b5ecdc80-b404-11eb-8dd5-b794a5d65e56.png)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.5.3, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202153-19-may-2021
